src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java

```diff
@@ -77,19 +85,29 @@ public void intakeStop() {
 	}
 
 	public GamePieceType detectType() {
-		// if () {
-		//     return GamePieceType.CUBE;
-		// }
-		// else if () {
-		//     return GamePieceType.CONE;
-		// }
-
+		if (colorSensor.getColor().equals(INTAKE_CUBE_COLOR)) {
+			return GamePieceType.CUBE;
+		} else if (colorSensor.getColor().equals(INTAKE_CONE_COLOR)) {
+			return GamePieceType.CONE;
+		}
 		return GamePieceType.NONE;
 	}
 
-	public boolean isSecured() { // Checks to see if the game piece is secured
+	public boolean
+			isSecured() { // Checks to see if the game piece is secured, returns true if the motor should
+		// stop
+		if (getDistance() < 12) { // units in cm
```

Comment:
It will be interesting to see how the intake reacts with cones not in expected orientations (tip during intake?)